dist/crawl_links.js

```diff
@@ -0,0 +1,45 @@
+/*
+Collects same-origin in-viewport links from img, h[1-6] tags,
+sorted by size from largest to smallest.
+This is the seed for generating a list of URLs to crawl from the given page.
+*/
+const intersectRect = function (r1, r2) {
+    return !(r2.left > r1.right || 
+              r2.right < r1.left || 
+              r2.top > r1.bottom ||
+              r2.bottom < r1.top);
+};
+const sameOrigin = function(uri1, uri2){
+    uri1 = new URL(uri1);
+    uri2 = new URL(uri2);
+    if(uri1.host !== uri2.host) return false;
+    if(uri1.port !== uri2.port) return false;
+    if(uri1.protocol !== uri2.protocol) return false;
+    return true;
```

Comment:
Can we simplify this?
```suggestion
    return uri1.origin == uri2.origin;
```